File: atcoder/1/C.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    ios::sync_with_stdio();
    int n, m; cin >> n >> m;
    array<vector<int>, 300100> adj;
    vector<int> ric(m);
    for (int i = 0; i < m; i++)
    {
        int k; cin >> k;
        ric[i] = k;
        for (int j = 0; j < k; j++)
        {
            int bj; cin >> bj;
            adj[bj].push_back(i);
        }
    }

    int cnt = 0;
    for (int i = 0; i < n; i++)
    {
        int b; cin >> b;
        for (auto p : adj[b])
        {
            ric[i]--;
            if (ric[i] == 0)
            {
                cnt++;
            }
        }
        cout <<  cnt << endl;
    }
}

File: atcoder/1/C.py
n, m = map(int, input().split())

adj = dict()

ric = [0] * m

for i in range(m):
    inp = list(map(int, input().split()))
    ric[i] = inp[0]
    for ig in inp[1:]:
        if not ig in adj.keys():
            adj[ig] = []
        adj[ig].append(i)

cnt = 0

tols = list(map(int, input().split()))
for to in tols:
    for i in adj[to]:
        ric[i] -= 1
        if ric[i] == 0:
            cnt += 1
    print(cnt)